shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -65,9 +66,9 @@ const strSdkNotInitialized = "SDK is not initialized";
  */
 const defaultConfig: IConfigDefaults<IConfiguration> = objDeepFreeze({
     cookieCfg: {},
-    [STR_EXTENSIONS]: [],
-    [STR_CHANNELS]: [],
-    [STR_EXTENSION_CONFIG]: {},
+    [STR_EXTENSIONS]: { rdOnly: true, ref: true, v: [] },
```

Comment:
This is one of the main parts that makes the `extensions` and `channels` readonly enforced.